src/main/java/eloquent/adapters/LocalAdapter.java

```diff
@@ -117,8 +144,21 @@ public File read(String path) throws EloquentException {
 
     @Override
     public Metadata getMetadata(String path) throws EloquentException {
-        // @TODO
-        return null;
+
+        path = this.buildPath(path);
+        java.io.File file = new java.io.File(path);
+
+        // File doesn't exist
+        if (!file.exists()) {
+            throw new FileNotFoundException(path);
+        }
+
+        Metadata metadataModel = new Metadata();
+
+        metadataModel.setName(file.getName())
+                    .setPath(file.getPath());
```

Comment:
@dhruvdutt are you able to get the size and timestamp (last modified date) of the file?